projects/delayed-execution-reth-lighthouse.md

```diff
@@ -0,0 +1,167 @@
+    //  to be used when validating the next block's headers
+}
+```
+
+5. `EngineApi` trait mods:
+- Add `new_payload_v5`
+    - accepts `ExecutionPayloadV4` with the additional fields for block validation.  We will have to modify the payload status to be returned after static validation.
+- `engine_getPayloadV5`
+    - returns `ExecutionPayloadV4` with new fields for block building
+- extend `EngineTypes` to define a new`ExecutionPayloadEnvelopeV5` and related types. 
+- add `EngineApiMessageVersion::V5` for the new hardfork.
+
+6. Structs like `ForkTracker` will need to have a placeholder hard fork property for use as to apply the logic above only after the fork boundary. 
+
+### CL Changes (Lighthouse)
+1.  The `ExecutionPayload` struct is included in the `BeaconBlockBody`, so we will need to add a new variant to apply past the fork boundary that will include our new fields.
+```
+enum ExecutionPayload<E: EthSpec> {
+    Eip7886(ExecutionPayloadEip7886<E>), // new variant
+}
+```
+
+2.  Additionally, the evergreen `BeaconState` will need to be updated past the fork boundary to have its `ExecutionPayloadHeader` hold the new `parent_*` fields.
+
+3.  The EL's block header is RLP encoded and then hashed to obtain the`block_hash`.  Since `block_hash` is stored in our `ExecutionPayload`, we'll need an `ExecutionBlockHeader` variant that includes the new fields.  The `block_hash` is later recalculated by the EL to check for validity.  The `block_hash` will also be persisted in the `BeaconState::ExecutionPayloadHeader`.
+
+4.  Lighthouse will JSON serialize the `ExecutionPayload` to send to the EL via the engine api for block validation.  Therefore, we'll need a `new_payload_eip7886` helper to forward the payload to the engine.  Similarly, a `get_payload_eip7886` helper will return payloads for block building.
+
+5.  Validator clients can request blinded blocks where only a payload header is sent. We'll need to extend this header:
+```rust
+pub struct BlindedPayload<E: EthSpec> {
+    execution_payload_header_eip7886: ExecutionPayloadHeaderEip7886<E>,
+    // existing variants
+}
+```
+
+With these updates, Lighthouse will properly store parent execution output fields in the `BeaconState` and the `BeaconBlockBody`.  
+
+## Roadmap
+
+| Period                    | Goals                                                                                              |
+|---------------------------|-----------------------------------------------------------------------------------------------------|
+| July – mid August         |  Lighthouse - update `BeaconState` and `BeaconBlockBody` execution payloads     |
+| Mid August – Mid October|  Header struct + engine api updates.  Then, update `ExecutionStage` + live sync path |
+| Mid October – DevConnect   | Testing and benchmarking of proposed changes     |
+
+## Possible challenges
+- Maintaining backward compatibility pre-fork boundary will be complex
```

Comment:
Can you elaborate on this? Seems like a short dismissal of an important issue. Also other potential challenges to consider - like delayed execution getting buried in research phase 🙃

Reply:
> Can you elaborate on this? Seems like a short dismissal of an important issue. Also other potential challenges to consider - like delayed execution getting buried in research phase 🙃

@taxmeifyoucan, since epbs was selected for glamsterdam, looks like delayed execution will be delayed indefinitely ha

Luckily, the Lighthouse team asked me to help out with EPBS some time ago (as we discussed), which is a pretty exciting opportunity.   I've made solid progress on it in the past couple weeks.  

I have the epbs project proposal complete on my end.  Once I get feedback from mentor, I'll send it your way for review, should be this week! In the meantime, I'll keep shipping